rfc#00003

```diff
@@ -0,0 +1,213 @@
+we assume that the Orchestrator is operated by the tenant and that the
+Tenant workload image repository is also operated by the tenant.
+
+#### State 0 non-Enarx-enabled use case (initial)
+No Enarx-related components are relevant to this use case.
+- host owner trust domain
+  - host
+  - host CPU + CPU firmware
+  - Other workloads
+- tenant trust domain
+  - Orchestrator
+  - Tenant workload image
+  - Tenant workload image repository
+
+Note - there are models where a trust relationship is maintained by
+the tenant to the Tenant workload image repository or to the tenant
+workload, but it is not directly within the trust domain.  These
+models are not examined in this document.
+
+#### State 1 non-Enarx-enabled use case (provisioned)
+- Host owner trust domain
+  - Host
+  - Host CPU + CPU firmware
+  - Tenant workload
+  - Other workloads
+- Tenant trust domain
+  - Orchestrator
+  - Tenant workload image
+  - Tenant workload image repository
+The change from state 0 to 1 is the provisioning of the Tenant workload
+onto the host from the Tenant workload image.
+
+The key point of this use case from a trust domain point of view it that
+since the host has access to all processes and associated memory that are
+running on it, all workloads that run on the host can only exist within
+the host trust domain.
+
+### Enarx use case states
+Note that exact state details may vary by implementation.
+
+#### State 0 Enarx use case (initial state)
+Initial state (we assume no compromised processes on host):
+- Host owner trust domain
+  - Host
+  - Host CPU + CPU firmware
+  - Enarx host agent
+  - Enarx runtime image
+  - Other workloads
+- Tenant trust domain
+  - Orchestrator
+  - Enarx client agent
+  - Tenant workload image
+  - Tenant workload image repository
+This is the same as the initial state for the non-Enarx-enabled
+use case, with the exception of the Enarx components.
+
+#### State 1 Enarx use case (TODO: check name - pre-attestation)
+- Host owner trust domain
+  - Host
+  - Host CPU + CPU firmware
+  - Enarx host agent
+  - TEE instance
+  - Enarx runtime image (within TEE instance)
+  - Other workloads
+- Tenant trust domain
+  - Orchestrator
+  - Enarx client agent
+  - Tenant workload image
+  - Tenant workload image repository
+The state change from 0 to one creates a TEE instance and loads
+the Enarx runtime image into it.  
+
+#### State 2 Enarx use case (TODO: check name - attested)
+- Host owner trust domain
+  - Host
+  - Host CPU + CPU firmware
+  - Enarx host agent
+  - Other workloads
+- tenant trust domain
+  - Orchestrator
+  - Enarx client agent
+  - Empty Keep
+  - Tenant workload image
+  - Tenant workload image repository
+The state change from 1 to 2 moves the Empty Keep into the tenant
+trust domain.  The TEE instance has transformed into a Keep as
+it has been loaded with the Enarx runtime agent and attested.
+
+#### State 3 Enarx use case (TODO: check name - provisioned)
```

Comment:
```suggestion
#### Enarx use case: state 3 (TODO: check name - provisioned)
```

Reply:
No. The state numbering is most important.